database/seeder.js

```diff
@@ -0,0 +1,99 @@
+
+/********  SAMPLE MOVIE DATA  ********/
+const titles = ['Captain America', 'Forrest Gump', 'Gone With The Wind', 'Bird Cage', 'The Lord Of The Rings', 'Harry Potter And The Order Of Phoenix', 'Get Smart', 'Die Hard', 'Mission Impossible', 'The Matrix', 'The Martian', 'Interstellar', 'It Might Get Loud', 'Spectre', 'Fat Albert', 'Monty Python And The Holy Grail', 'Die Another Day', 'The Happening', 'Beauty And The Beast', 'Gran Torino', 'Apollo 13', 'War Of The Worlds', 'Hancock', 'Independence Day', `Ender's Game`, 'The Fugitive', 'Birdman', 'Jurassic Park', 'Star Wars: A New Hope', 'The Hobbit: An Unexpected Journey', 'Men In Black'];
+
+const dates = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+
+
+/*********** HELPER FUNCTIONS ***********/
+const getRandomNum = (max) => {
+  return Math.floor(Math.random() * max);
+}
+
+const generatePersonnelForMovies = (max) => {
+  let results = [];
+  for (let i = 0; i <= getRandomNum(13); i++) { //each movie will have up to 13 cast/crew members
+    results.push({ id: getRandomNum(max), role: roles[getRandomNum(19)] })
+  }
+  return results;
+}
+
+/********** OBJECT GENERATOR *************/
+let fakeMovie = () => {
+  return {
+    title: titles[getRandomNum(31)],
+    release_date: `${dates[getRandomNum(12)]} ${getRandomNum(27) + 1}`,
+    vudu_rating: getRandomNum(5) + 1,
+    rt_rating: getRandomNum(101),
+    price: `$${getRandomNum(30) + 1}.99`,
+    thumbnail_url: `https://mapquiz.app/fec/thumbnails/movie_thumbnail${getRandomNum(46) + 1}.jpeg`,
+    personnel: generatePersonnelForMovies(21)
+  }
+};
+
+let fakePerson = (actor, movieList) => {
+  let person = {
+    name: actor,
+    thumbnail_url: `https://mapquiz.app/fec/headshots/headshot${getRandomNum(15) + 1}.jpeg`,
+    movies: []
+  }
+
+  for (let i = 0; i <= getRandomNum(31); i++) { //each cast/crew member will have up to 31 movies
+    person.movies.push(movieList[getRandomNum(31)]);
+  }
+
+  return person;
+};
+
+/******* DATABASE SEEDER *******/
+/* NEEDS REFACTORING! Todo: Kill the mongoose connection once fully seeded w/o interfering with Jest tests */
+
+const seed = () => {
+  return new Promise((resolve, reject) => { //we want async nature for our tests
+
+    for (let i = 1; i <= 100; i++) {
+      const currentMovie = new db.Movie(fakeMovie());
+      currentMovie.save().then(() => {
+        bar.update(++completed);
+        completed === 100 && generateMoviesForPersonnel();
+      });
+
+    }
+
+    const generateMoviesForPersonnel = () => {
+      db.Movie.find().distinct('_id').then((movieList) => { //gets random movie IDs to assign to personnel
+
+        for (let actor of actors) {
+          let currentPerson = new db.Personnel(fakePerson(actor, movieList));
+          currentPerson.save().then(() => {
+            bar.update(++completed);
+            if (completed === 122) {
+              bar.stop()
+              console.log('Finished seeding!');
+              resolve();
```

Comment:
What's going on w/ the code here? Do we really need to use `resolve()` as a closure variable to execute it?

Reply:
@therobinkim, thank you for the great review! Incorporating your suggestions now.